doc/modules/language-guide/pages/upgrades.adoc

```diff
@@ -118,6 +106,70 @@ Note that the type of `+entries+`, being just an array of `+Text+` and `+Nat+` p
 In this example, the `+preupgrade+` system method simply writes the current `+map+` entries to `+entries+` before `+entries+` is saved to stable memory.
 The `+postupgrade+` system method resets `+entries+` to the empty array after `+map+` has been populated from `+entries+` to free space.
 
-== Upgrading the compiled program
+== Stable type signatures
+
+The collection of stable variable declarations in an actor can be summarized in a
+_stable signature_.
+
+The textual representation of an actor's stable signature resembles the internals of a {proglang} actor type:
+
+[source.no-repl, motoko]
+....
+actor {
+  stable x : Nat;
+  stable var y : Int;
+  stable z : [var Nat];
+};
+....
+
+It specifies the names, types and mutability of the actor's stable fields, possibly preceded by relevant Motoko type declarations.
+
+TIP: You can emit the stable signature of the main actor or actor class to a `.most` file using `moc` compiler option `--stable-types`. You should never need to author your own `.most` file.
+
+A stable signature, `<stab-sig1>`, is _stable-compatible_ with signature `<stab-sig2>`, if, and only,
+
+* every immutable field `stable <id> : T` in `<stab-sig1>` has a matching field `stable <id> : U` in `<stab-sig2>` with `T <: U`.
+* every mutable field `stable var <id> : T` in `<stab-sig1>` has a matching field `stable var <id> : U` in `<stab-sig2>` with `T <: U`.
+
+Note that `<stab-sig2>` may contain additional fields.
+
+The subtyping condition on stable fields ensures that the final value of some field can be consumed as the initial value of that field in the upgraded code.
+
+TIP: You can check the stable-compatiblity of two `.most` files, `cur.most` and `nxt.most` (containing stable signatures), using `moc` compiler option `--stable-compatible cur.most nxt.most`.
+
+NOTE: The _stable-compatible_ relation is quite conservative. In future, it may be relaxed to accommodate a change in field mutability
```

Comment:
```suggestion
NOTE: The _stable-compatible_ relation is quite conservative. In the future, it may be relaxed to accommodate a change in field mutability
```